README.md

```diff
@@ -510,41 +510,40 @@ The price of burger is 10$
 ##### `$>` `$<` increases/decreases the indentation level.
 
 ```java
-CodeBlock codeBlock =CodeBlock.builder().add("CodeBlock codeBlock =CodeBlock.builder()")
-                .add("$>$>")
-                .add("\n.add($S, $S)", '\'', "&#39;")
-                .add("\n.add($S, $S)", '>', "&gt;")
-                .add("\n.build()")
-                .add("$<$<")
+CodeBlock codeBlock =CodeBlock.builder().add("class Student")
+                .add("{$>$>\n")
+                .add("int studentId;\n")
+                .add("String name;\n")
+                .add("$<$<}")
                 .build();
 ```
 
 ```java
-CodeBlock codeBlock =CodeBlock.builder()
-    .add("'", "&#39;")
-    .add(">", "&gt;")
-    .build()
+class Student{
+    int studentId;
+    String name;
+}
 ```
 
 ##### `$[` `$]` begin/end a statement. 
 
 `$[` and `$]` are used to start or end a statement. What should be mentioned is that every line after the first line will be double-indented if there is multiple statements.
 
 ```java
-CodeBlock codeBlock =CodeBlock.builder().add("class Student")
+CodeBlock codeBlock =CodeBlock.builder().add("void method")
                 .add("{$[\n")
-                .add("String name;\n")
-                .add("int score;\n")
-                .add("public Student(){}")
-                .add("$]\n}")
+                .add("int a = 1;\n")
+                .add("int b = 2;\n")
+                .add("int c = a+b;\n")
+                .add("$]}")
```

Comment:
Those are 3 statement within a (bogus) method declaration. Basically statement ends with `;`  (this is far from being that simple, but it's a good approximation). Your initial indent example **was** a single statement on multiple lines, and is exactly what those placeholders are meant to make easier (compared to using the indent placeholders).

```java
CodeBlock.builder().
        .add("$[$1T foo = $1T.builder()\n", SomeType.class)
        .add(".foo($S)\n", "bar")
        .add(".baz(42)\n")
        .add(".build();\n$]");
```

See examples in https://github.com/square/javapoet/blob/304176ef98fa80f35cb49ed130620c7b0ecdf5a6/src/test/java/com/squareup/javapoet/TypeSpecTest.java#L2359-L2408 and https://github.com/square/javapoet/blob/304176ef98fa80f35cb49ed130620c7b0ecdf5a6/src/test/java/com/squareup/javapoet/TypeSpecTest.java#L1632-L1719 (because `addStatement` will use `$[` and `$]`)

Reply:
do you mean that I shouldn't put a newline after $[, I am confused